content/flutter-publishing/publishing-to-app-store.md

```diff
@@ -95,3 +97,15 @@ To add localized release notes that will appear in the Test Details (What to tes
 ```
 
 Supported languages could be found [here](https://developer.apple.com/documentation/appstoreconnectapi/betabuildlocalizationcreaterequest/data/attributes).
+
+Note: Uploading release notes takes place in the [post-processing](#post-processing-of-app-store-connect-distribution) step.
+
+## Post-processing of App Store Connect distribution
+
+Some App Store Connect actions, like submitting the build to TestFlight beta review, disrtributing the build to beta groups and uploading release notes take place asynchronously in the post-processing step after the app artifact has been successfully published to App Store Connect and the main workflow has completed running in Codemagic. This avoids using the macOS build machine while we are waiting for Apple to complete processing the build and it becomes available for further actions. 
+
+Post-processing has a two-step timeout. If the uploaded build cannot be found in App Store Connect after 15 minutes, it is likely that Apple has detected an issue with the artifact and post-processing is cancelled. In total, post-processing runs for 60 minutes maximum. If the uploaded build has not exited the processing status by then, post-processing in cancelled. You will be still able to manually submit the build to beta review, upload release notes and distribute the app to beta groups once the build becomes available in App Store Connect.
```

Comment:
Maybe add few more words about the initial 15 min timeout. Explain that in case Apple has detected an issue, then the build information will not become available at all and as such we're unable to provide any kind of information.